packages/yarnpkg-fslib/sources/buffer.ts

```diff
@@ -0,0 +1,29 @@
+import {NativePath, PortablePath, Path, PathUtils, PathLike, PathType, ppath, npath} from './path';
+
+export class BasePathBuffer<P extends Path> extends Buffer {
+  declare public toString: (encoding?: string, start?: number, end?: number) => P;
+
+  protected constructor(pathUtils: PathUtils<P>, p: PathLike<P>, encoding?: BufferEncoding) {
+    super(pathUtils.fromPathLike(p), encoding);
+  }
+}
+
+export class PortablePathBuffer extends BasePathBuffer<PortablePath> {
+  declare _path_type: PathType.Portable | PathType.File;
```

Comment:
Why `declare`?

Reply:
Because `_path_type` is just a brand used for the nominal typing, it wouldn't make sense to initialize it.

Reply:
Ok - and why `PathType.File`? I only see two references to that in the diff

Reply:
You're right - it's not needed, since there's no `Filename` equivalent for `PathBuffer`s. I'll remove it 👍